crates/rooch-key/src/key_derive.rs

```diff
@@ -148,12 +170,14 @@ pub fn generate_new_key_pair<Addr, KeyPair>(
     key_pair_type: KeyPairType,
     derivation_path: Option<DerivationPath>,
     word_length: Option<String>,
+    password: Option<String>,
 ) -> Result<(Addr, KeyPair, KeyPairType, String), anyhow::Error>
 where
     KeyPairType: CoinOperations<Addr, KeyPair>,
 {
     let mnemonic = Mnemonic::new(parse_word_length(word_length)?, Language::English);
-    let seed = Seed::new(&mnemonic, "");
+    let password_str = password.as_deref().unwrap_or("");
+    let seed = Seed::new(&mnemonic, password_str);
```

Comment:
I think using the `password_str` as part of the seed is not a good idea for CLI.

1. The user needs to save the mnemonic and password and use the mnemonic and password to recover the account. If use a wrong password, will recover a new account.
2. The user can not change the password.

Some wallet provides a `salt` to mnemonic, like this, and the user needs to know it is salt, not the password.

Reply:
> The user needs to save the mnemonic and password and use the mnemonic and password to recover the account. If use a wrong password, will recover a new account.

This is a common practice for hardware and software authentication schemes e.g. MetaMask with mnemonic and password, or BitBox02 with password option. Here, password is merely an option for the user and the user can opt in to choose whether to remember it.

> The user can not change the password.

If changing password would generate a new account, it is the common practice for the bitcoin software bc we are using bip39 mnemonic word Rust lib. In other words, changing password would mean collapsing the existing seed infrastructure at the first time generated. There's an article introducing the passphrase trade-offs: https://help.blockstream.com/hc/en-us/articles/8712301763737-What-is-a-BIP39-passphrase-#h_01G8K3SK5VBHBYZSX5XS4309YG.

I think it's only a choice made by a user. 

> Some wallet provides a salt to mnemonic, like this, and the user needs to know it is salt, not the password.

The accurate answer is passphrase: https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki#user-content-From_mnemonic_to_seed. Salt is from a password with mnemonic string according to the original code:

```rs
let salt = format!("mnemonic{}", password);
```

We can add documents about how the seed phrase is constructed.